src/core/positioning/internalgnssreceiver.cpp

```diff
@@ -177,7 +177,7 @@ void InternalGnssReceiver::handleSatellitesInUseUpdated( const QList<QGeoSatelli
                                                               mSatellitesInfo, 0, 0, 0,
                                                               mLastGnssPositionInformation.hacc(),
                                                               mLastGnssPositionInformation.vacc(),
-                                                              mLastGnssPositionInformation.utcDateTime(),
+                                                              mLastGnssPositionInformation.utcDateTime().isValid() ? mLastGnssPositionInformation.utcDateTime() : QDateTime::currentDateTimeUtc(),
```

Comment:
This code is repeated a bunch of times.  That makes me wonder:
  - should it be a procedure?
  - why check if datetime is not valid, but use hacc/vacc unconditionally?   Is there really a situation with vllid accuracy but not valid time?

It seems clear something tricky is going on, but I don't understand it from the comments you didn't add, and I don't understand it from the commit message.     Maybe GNSS data should not be used at all if the associated datetime is not valid, and that's the real bug.